### ISSUE TITLE
BODY becomes child of HEAD when closing tag is missing

### ISSUE THREAD

**Issue by arrix** [tautologistics/node-htmlparser#12]
Some old W3C pages were written in old style HTML. e.g. http://www.w3.org/TR/css3-2d-transforms/ 

node-htmlparser should be more forgiving.
version: 1.7.2

```
var request = require('request');
var jsdom = require('jsdom');

var url = 'http://www.w3.org/TR/css3-2d-transforms/';
request({uri:url}, function (error, response, body) {
    var html = body;
    var doc = jsdom.jsdom(html, null, {url: url});
    console.log(doc.head+''); //[ HEAD ]
    console.log(doc.body === null); //true
    console.log(doc.head.childNodes[9].tagName); //BODY
});
```

**Comment by tautologistics** [comment]
Yeah, there are actually quite a few closing tags that can be omitted per spec. I have something working and will roll it into some planned 1.8.0 changes.

**Comment by eirikurn** [comment]
Ahh, scraping badly coded pages can be a pain. 

I just wasted 2 hours trimming a bad page to find out what was causing strange DOM errors, and finally found that htmlparser is not forgiving for pages that omit a closing `tr` tag. There were no `tr` tags closed in the page I was scraping, so the complete table was a huge recursive `tr`.

I look very forward to 1.8.0 if it fixes this :)
